src/funnels/automation/StableSwapper.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface SwapperLike {
+    function swap(address, address, uint256, uint256, address, bytes calldata) external returns (uint256);
+}
+
+contract StableSwapper {
+    mapping (address => uint256) public wards;                           // Admins
+    mapping (address => uint256) public buds;                            // Whitelisted keepers
+    mapping (address => mapping (address => PairConfig)) public configs; // Configuration for keepers
+
+    SwapperLike public immutable swapper;                                // Swapper for this StableSwapper
+
+    struct PairConfig {
+        uint128 num; // The remaining number of times that a src to dst swap can be performed by keepers
+        uint32  hop; // Cooldown period it has to wait between swap executions
+        uint32  zzz; // Timestamp of the last swap execution
+        uint96  lot; // The amount swapped by keepers from src to dst every hop
+        uint96  req; // The minimum required output amount to insist on in the swap form src to dst
```

Comment:
`num` could just be `uint256` here since the last four fields pack into a full slot anyway (should save a little gas on bit manipulation).

Reply:
If we do that then the storage writes of `num` and `zzz`, which are done on each `swap` operation, will become two writes (to 2 storage slots) instead of one with bit manipulation.

Reply:
Ah okay...in that case you could expand `lot` and `req` to 128 bits but I guess you don't want to do that for consistency reasons.